### PR TITLE
Updated event filtering

### PR DESCRIPTION
**Pull request by KrasuliaV**

### CODE REVIEW

**Review by Helen1711 — APPROVED** [review]

File: sandbox-data/src/main/java/com/exadel/sandbox/repository/event/specification/SpecificationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import java.util.Set;
+
+@Repository
+@RequiredArgsConstructor
+public class SpecificationBuilder {
+
+    private final EventRepository eventRepository;
+
+    public Page<Event> getEventsByParameters(Status status,
```
Comment (AndriiHolovianko) [inline sandbox-data/src/main/java/com/exadel/sandbox/repository/event/specification/SpecificationBuilder.java:20]: If you have more than 3 parameters consider refactoring or wrapper object creation.
Effective Java, Chapter 7 (Methods), Item 40 (Design method signatures carefully)

File: sandbox-service/src/main/java/com/exadel/sandbox/service/impl/EventServiceImp.java [hunk 2]
```diff
@@ -109,182 +113,23 @@ public EventDetailsResponse getEventById(Long eventId) {
...
+                eventFilterRequest.getTagsIdSet(),
+                eventFilterRequest.getVendorsIdSet(),
+                PageRequest.of(pageNumber, pageSize, sort));
 
-        } else {
-            eventPage = eventRepository.findByTagsByCountry(eventFilterRequest.getLocationId(),
-                    eventFilterRequest.getTagsIdSet(),
-                    eventFilterRequest.getStatus(),
-                    PageRequest.of(pageNumber, pageSize, sort));
+        return eventFilterRequest.isCity() ?
```
Comment (AndriiHolovianko) [inline sandbox-service/src/main/java/com/exadel/sandbox/service/impl/EventServiceImp.java:125]: The ternary operator may impair readability, maybe if-else will be better here

File: sandbox-data/src/main/java/com/exadel/sandbox/repository/event/EventRepository.java [hunk 3]
```diff
@@ -6,29 +6,25 @@
...
 
 import javax.transaction.Transactional;
-import java.util.Set;
 
 @Repository
-public interface EventRepository extends JpaRepository<Event, Long>, EventRepositoryCustom {
+public interface EventRepository extends JpaRepository<Event, Long>, EventRepositoryCustom, JpaSpecificationExecutor<Event> {
+
+    /*TODO To determine what method are the best*/
+//    Page<Event> findDistinctByLocationsCityIdAndStatus(Long cityId, Status status, Pageable pageable);
```
Comment (AndriiHolovianko) [inline sandbox-data/src/main/java/com/exadel/sandbox/repository/event/EventRepository.java:20]: Do you need this todo statement?

File: sandbox-data/src/main/java/com/exadel/sandbox/repository/event/specification/SpecificationBuilder.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
+package com.exadel.sandbox.repository.event.specification;
```
Comment (AndriiHolovianko) [inline sandbox-data/src/main/java/com/exadel/sandbox/repository/event/specification/SpecificationBuilder.java:1]: Great work! Well done, now it looks much better:)

**Review by RosKokos — APPROVED** [review]

File: sandbox-data/src/main/java/com/exadel/sandbox/repository/event/specification/SpecificationBuilder.java (hunk 1 above)
Comment (KrasuliaV) [reply sandbox-data/src/main/java/com/exadel/sandbox/repository/event/specification/SpecificationBuilder.java:20]: Yes, I understand this problem.
But EventFilterRequest is from the service level.
Will it be right if I pass this object as a method parameter from the service to the repo?

**Pull request merged**